scIB/metrics.py

```diff
@@ -1667,22 +1667,28 @@ def kBET(adata, batch_key, label_key, embed='X_pca', type_ = None,
 
 # determine root cell for trajectory conservation metric
 def get_root(adata_pre, adata_post, ct_key, dpt_dim=3):
+    
     n_components, adata_post.obs['neighborhood'] = connected_components(csgraph=adata_post.uns['neighbors']['connectivities'], directed=False, return_labels=True)
     
     start_clust = adata_pre.obs.groupby([ct_key]).mean()['dpt_pseudotime'].idxmin()
-    min_dpt = np.flatnonzero(adata_pre.obs[ct_key] == start_clust)
-    max_neigh = np.flatnonzero(adata_post.obs['neighborhood']== adata_post.obs['neighborhood'].value_counts().argmax())
+    min_dpt = adata_pre.obs[adata_pre.obs[ct_key] == start_clust].index
+    #print(min_dpt)
+    max_neigh = adata_post.obs[adata_post.obs['neighborhood']== adata_post.obs['neighborhood'].value_counts().idxmax()].index
     min_dpt = [value for value in min_dpt if value in max_neigh]
     
+    adata_post_sub = adata_post[adata_post.obs['neighborhood']== adata_post.obs['neighborhood'].value_counts().argmax()]
```

Comment:
if argmax is deprecated and idxmax should be used instead, why not also here?

Reply:
could also save `.value_counts().idxmax()` somewhere, as you reuse it here.

Reply:
ah, I missed that one

Reply:
I realised this actually breaks the function if the largest component isn't 0

Reply:
isn't the largest component always 0? I think it should be ordered... but better to not make assumptions i guess.